keps/sig-cluster-lifecycle/wgs/20200108-instance-specific-componentconfig.md

```diff
@@ -0,0 +1,418 @@
+auto-detect instance specific config in the future. It is worth noting that the
+_possibility_ of someday being able to auto-detect these has led to some
+hesitation to make a firm decision today. We have delayed for too long already,
+and a firm decision is needed to move forward.
+
+### Risks and Mitigations
+
+This creates an additional bucket for config parameters. We must be vigilant
+that only parameters that are truly instance-specific are added to the
+instance-specific Kind. We can provide warnings to developers via comments
+and tests to deter adding parameters to the instance-specific Kind that should
+instead be in the sharable Kind.
+
+For users deploying Pod templates, this solution can make the template more
+verbose. In the future, we can explore more elegant ways of generating files
+with instance-specific parameters correctly substituted. This proposal argues
+that the extra verbosity is worth the benefit of a consistent, versioned API for
+configuring core components.
+
+## Design Details
+
+ComponentConfig APIs are currently defined as a separate API group for each
+component, usually containing a single top-level Kind for configuring that
+component. The naming convention for the API group is `{component}.config.k8s.io`,
+and the convention for the Kind is `{Component}Configuration`. This KEP proposes
+that the naming convention for the new Kind be `{Component}InstanceConfiguration`.
+
+The canonical flag for providing a ComponentConfig file to a component is
+`--config`. This KEP proposes that the canonical flag for providing an
+instance-specific config file be `--instance-config`, and that the
+instance-specific object not initially be permitted as part of a yaml stream
+in the `--config` file (and vice-versa). This is for the sake of a simple
+implementation and can be enabled in the future, if we decide it is useful.
+
+As with sharable ComponentConfig parameters, command line flags for
+instance-specific parameters should continue to function and take precedence
+over values from the config file so that backwards compatibility is maintained.
+
+### Kube-proxy Example
+
+Today, a kube-proxy DaemonSet may be configued as follows to mix sharable and
+instance-specific configuration parameters (using flags only). Note some fields
+and arguments that would be set in production have been removed or modified
+below to improve the clarity of the example.
+
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+  name: kube-proxy
+  namespace: kube-system
+spec:
+  template:
+    spec:
+      containers:
+      - name: kube-proxy
+        image: kube-proxy-amd64
+        command:
+        - kube-proxy
+        - --proxy-mode=ipvs
+        - --hostname-override=${NODE_NAME}
+        env:
+          - NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+```
+
+Below, how the above may be modified when moving to a shared ComponentConfig
```

Comment:
I think you're missing a word here: `Below demonstrates how the above may be modified ...`